### PR TITLE
Add PSScriptAnalyzer workflow

### PR DESCRIPTION
**Pull request by mazhelez**

### CODE REVIEW

File: .github/workflows/powershell.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+      actions: read # only required for a private repository by github/codeql-action/upload-sarif to get the Action run status
```
Comment (mazhelez) [inline .github/workflows/powershell.yaml:28]:
```suggestion
```

Not needed.

File: .github/workflows/powershell.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - uses: actions/checkout@v3
+
+      - name: Run PSScriptAnalyzer
+        uses: microsoft/psscriptanalyzer-action@6b2948b1944407914a58661c49941824d149734f
+        with:
+          # Check https://github.com/microsoft/action-psscriptanalyzer for more info about the options.
+          # The below set up runs PSScriptAnalyzer to your entire repository and runs some basic security rules.
+          path: .\
+          recurse: true
+          # Include your own basic security rules. Removing this option will run all the rules
```
Comment (mazhelez) [inline .github/workflows/powershell.yaml:41]:
```suggestion
```

Comment not needed

File: .github/workflows/powershell.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          includeRule: '"PSAvoidGlobalAliases", "PSAvoidUsingConvertToSecureStringWithPlainText"'
```
Comment (mazhelez) [inline .github/workflows/powershell.yaml:42]: We probably need more rules.

Comment (mazhelez) [reply .github/workflows/powershell.yaml:40]:
```suggestion
          excludeRule: '"PSAvoidUsingInvokeExpression", "PSUseShouldProcessForStateChangingFunctions", "PSAvoidUsingWriteHost", "PSAvoidUsingCmdletAliases"'
```

File: .github/workflows/powershell.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    permissions:
+      contents: read # for actions/checkout to fetch code
+      security-events: write # for github/codeql-action/upload-sarif to upload SARIF results
+    name: PSScriptAnalyzer
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run PSScriptAnalyzer
+        uses: microsoft/psscriptanalyzer-action@6b2948b1944407914a58661c49941824d149734f
```
Comment (aholstrup1) [inline .github/workflows/powershell.yaml:34]: Why from that specific commit? Why not '@main' or '@v1.1'?

File: .github/workflows/powershell.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+name: PSScriptAnalyzer
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  schedule:
+    - cron: '00 01 * * 2'
```
Comment (aholstrup1) [inline .github/workflows/powershell.yaml:18]: I wonder if need to run it on a schedule AND on push? On push + on pull requests should be fine no?

File: .github/workflows/powershell.yaml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    permissions:
+      contents: read # for actions/checkout to fetch code
```
Comment (aholstrup1) [inline .github/workflows/powershell.yaml:26]: Isn't this permission inherited from the permissions in line 20-21? Or does GitHub not work that way? :D

Consider just moving the security-events permission after line 21 and removing the permissions from the _build_ job

**Review by aholstrup1 — APPROVED** [review]

**Review by freddydk — APPROVED** [review]

File: .github/workflows/powershell.yaml [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Run PSScriptAnalyzer
+        uses: microsoft/psscriptanalyzer-action@6b2948b1944407914a58661c49941824d149734f
+        with:
+          # Check https://github.com/microsoft/action-psscriptanalyzer for more info about the options.
+          # The below set up runs PSScriptAnalyzer to your entire repository and runs some basic security rules.
+          path: .\
```
[the remark this answers is not in the document]
Comment (mazhelez) [reply .github/workflows/powershell.yaml:38]: Let's keep it like that for now.

File: .github/workflows/powershell.yaml (hunk 4 above)
Comment (mazhelez) [reply .github/workflows/powershell.yaml:34]: Use v.1.1: https://github.com/microsoft/psscriptanalyzer-action/tree/v1.1

```suggestion
        uses: microsoft/psscriptanalyzer-action@v1.1
```

File: .github/workflows/powershell.yaml (hunk 5 above)
Comment (mazhelez) [reply .github/workflows/powershell.yaml]: It's on push and pr now.

File: .github/workflows/powershell.yaml (hunk 6 above)
Comment (mazhelez) [reply .github/workflows/powershell.yaml]: Moved them.

**Review by aholstrup1 — APPROVED** [review]

**Pull request merged**